src/Microsoft.Extensions.SecretManager.Tools/Internal/ProjectIdResolver.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    public class ProjectIdResolver : IDisposable
+    {
+        private const string TargetsFileName = "FindUserSecretsProperty.targets";
+        private readonly ILogger _logger;
+        private readonly string _workingDirectory;
+        private Stack<Action> _disposables = new Stack<Action>();
+
+        public ProjectIdResolver(ILogger logger, string workingDirectory)
+        {
+            _workingDirectory = workingDirectory;
+            _logger = logger;
+        }
+
+        public string Resolve(string project, string configuration = Constants.DefaultConfiguration)
+        {
+            var finder = new MsBuildProjectFinder(_workingDirectory);
+            var projectFile = finder.FindMsBuildProject(project);
+
+            _logger.LogDebug(Resources.Message_Project_File_Path, projectFile);
+
+            var targetFile = GetTargetFile();
+            var outputFile = Path.GetTempFileName();
+            _disposables.Push(() => TryDelete(outputFile));
+
+            var commandOutput = new List<string>();
+            var commandResult = Command.CreateDotNet("msbuild",
+                new[] {
+                    targetFile,
+                    "/nologo",
+                    "/t:_FindUserSecretsProperty",
+                    $"/p:Project={projectFile}",
+                    $"/p:OutputFile={outputFile}",
+                    $"/p:Configuration={configuration}"
+                })
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .OnErrorLine(l => commandOutput.Add(l))
```

Comment:
Can't you just forward error and output to a string buffer? It'd save you from string.Joining below